docs/src/gui/gstat.adoc

```diff
@@ -170,429 +177,550 @@ class HandlerClass:
-intended to be sent when moving the cursor one line down in G-code display. +
-This depends on the widget/libraries used. +
-
-*dialog-request* :: '(returns python dict)' -
-intended to be sent when requesting a gui dialog. +
-It uses a python dict for communication. +
-The dict must include the following keyname pair: +
-* NAME: 'requested dialog name' +
-The dict usually has more keyname pair - it depends on the dialog. +
-dialogs return information using a general message +
-This depends on the widget/libraries used. +
-
-*focus-overlay-changed* :: '(returns bool, string, python object)' -
-intended to be sent when requesting an overlay to be put over the display. +
-This depends on the widget/libraries used. +
-
-*play-sound* :: '(returns string)' -
-intended to be sent when requesting a specific sound file to be played. +
-This depends on the widget/libraries used. +
-
-*virtual-keyboard* :: '(returns string)' -
-intended to be sent when requesting a on screen keyboard. +
-This depends on the widget/libraries used. +
-
-*dro-reference-change-request* :: '(returns integer)' -
-intended to be sent when requesting a DRO widget to change it's reference. +
-0 = machine, 1 = relative, 3 = distance-to-go +
-This depends on the widget/libraries used. +
-
-*show-preferences* :: '(returns None)' -
-intended to be sent when requesting the screen preferences to be displayed. +
-This depends on the widget/libraries used. +
-
-*shutdown* :: '(returns None)' -
-intended to be sent when requesting linuxcnc to shutdown. +
-This depends on the widget/libraries used. +
+==  Messages
 
-*error* :: '(returns integer, string)' -
-intended to be sent when an error has been reported . +
-integer represents the kind of error. ERROR, TEXT or DISPLAY +
-string is the actual error message. +
-This depends on the widget/libraries used. +
-
-*general* :: '(returns python dict)' -
-intended to be sent when message must be sent that is not covered by a more specific message. +
-General message should be used a sparsely as reasonable because all object connected to it will have to parse it. +
-It uses a python dict for communication. +
-The dict should include and be checked for a unique id  keyname pair: +
-* ID: 'UNIQUE_ID_CODE' +
-The dict usually has more keyname pair - it depends on implementation. +
-
-*forced-update* :: '(returns None)' -
-intended to be sent when one wishes to initialize or arbitrarily update an object. +
-This depends on the widget/libraries used. +
-
-*progress* :: '(returns integer, python object)'
-intended to be sent to indicate the progress of a filter program. +
-This depends on the widget/libraries used. +
-
-*following-error* :: '(returns python list)'
-returns a list of all joints current following error +
+*periodic*::
+  Returns nothing +
+  Sent every 100 ms.
+
+*state-estop*::
+  Returns nothing +
+  Sent when linuxcnc is goes into estop.
+
+*state-estop-reset*::
+  Returns nothing +
+  Sent when linuxcnc comes out of estop.
```

Comment:
You have done a lot of great improvements, but on these I don't agree. IMHO the original formatting was more clearly arranged.  i have reverted the corresponding lines on my branch.

Reply:
This is only a first pass actually.
Final idea is more like

``̀
*`hard-limits-tripped`*::
  * _Returns_: `bool, Python List` +
  * _Sent_: When any hard limit is tripped. +
  * _Comment_:
    * `bool` indicates if any limit is tripped,
    * `list` shows all available joint's current limit values.
```

Reply:
But the compact notation has also advantages...

This could be a compromise:

- state-estop 
_returns (nothing)_ - Sent when linuxcnc is goes into estop.

- hard-limits-tripped  
      _returns (bool, Python List)_ -  Sent when any hard limit is tripped.
     bool: indicates if any limit is tripped
     Python list: the list shows all available joint's current limit values

Reply:
> But the compact notation has also advantages...

Agreed, but I believe structure brings more :-P

What about:

```
*`hard-limits-tripped`* - `(bool, list)`::
  * _Event_: Any hard limit is tripped. +
  * _Return_:
    * `bool`: boolean indicating if any limit is tripped,
    * `list`: Python list with all available joint's current limit values.
```

Reply:
Looks good. But let's do this at a later stage. This should be basically aim at syncing, not improving the docs.

Reply:
The sync may include a superior idea for the formatting that the translator came up with. And since now is the last opportunity to avoid also affect the upcoming translations, I kind of like to see such discussions and would eventually frame the style guide.

Reply:
But this file has no counterpart in ES or FR, so it wouldn't affect translated files